is_core/utils/field_api.py

```diff
@@ -95,9 +96,12 @@ def get_widget(self):
             return ManyToManyReadonlyWidget
         elif field.auto_created and field.one_to_one:
             return ModelObjectReadonlyWidget
-        else:
+        elif isinstance(field, Field):
```

Comment:
Why not `elif hasattr(field, 'formfield')`?

Reply:
because this is more clear solution. Every real modle field is instance of Field. There is right implementation formfield. If somebody implements same method there should be returned False

Reply:
OK, I was just thinking if this method gets removed for some reason, this code will fail. But that's maybe not bad - you'll know something presumed is no longer true.

Reply:
yes if this method will be removed and than django fields api will be changed it will be better to raise exception